src/ScriptValidation.h

```diff
@@ -0,0 +1,77 @@
+// See the file "COPYING" in the main distribution directory for copyright.
+
+#pragma once
+
+#include "zeek/Traverse.h"
+
+namespace zeek::detail
+	{
+
+// Track statements and hooks and validate break and next usage.
+class BreakNextScriptValidation : public TraversalCallback
+	{
+public:
+	TraversalCode PreStmt(const Stmt* stmt)
+		{
+		stmt_depths[stmt->Tag()] += 1;
```

Comment:
Now this is tracking all statements, which semantically feels a bit odd. How about on more helper: `IsRelevantStatement()` (or such) testing whether it's a any of our stmts? And then just early-exit in all the methods if not relevant.